runtime/compiler/runtime/JITServerAOTCache.hpp

```diff
@@ -49,6 +51,8 @@ namespace TR { class Monitor; }
 // record types) in order to simplify memory management and exception handling.
 // The subclasses store their underlying serialization record data inline
 // (as well as variable-length arrays of subrecord pointers in some cases).
+// The subclasses also store a _nextRecord pointer that points to the next record
```

Comment:
Why are `_nextRecord` and `get/setNextRecord()` not simply members of the base class `AOTCacheRecord`?

Reply:
I was thinking about the same thing. `CachedAOTMethod` would need to be treated separately.
When walking the linked list we would get a generic `AOTCacheRecord` from `getNextRecord()`. This may require some downcasting when serializing to file, but maybe the cast can be avoided by virtual calls.

Reply:
I suppose that they could be, since we only care about the underlying `AOTSerializationRecord`.

Reply:
No downcasting is needed, there is already a virtual call for this - `AOTCacheRecord::dataAddr()`, which returns the `AOTSerializationRecord` (that contains its own size) that will be written to the file.